### PR TITLE
Experiment with testing `unittest` on Windows

### PR DESCRIPTION
**Pull request by wilzbach**
This will probably take a couple of hops again. Let's hope for the best...

### CODE REVIEW

**Comment by wilzbach** [comment]
```
Error on line 8: expecting target : dependencies
```

Looks like some patching error? :/

**Comment by wilzbach** [comment]
> Looks like some patching error? :/

Yep -> https://github.com/braddr/at-client/blob/master/src/diff-dmd-win64.diff
We should really get rid of these :/

**Comment by wilzbach** [comment]
This is blocked on https://github.com/braddr/at-client/pull/8 due to auto-tester's patching :/

**Comment by wilzbach** [comment]
This is now failing with:

```
..\generated\windows\release\32\dmd.obj(dmd) 
 Error 42: Symbol Undefined ?out_config_debug@@YAX_N000000@Z
Error: linker exited with status 1
```

Has anyone seen this before?

**Comment by jacob-carlborg** [comment]
Unless you already figured this out, `out_config_debug` is defined in `dmd/backend/backconfig.c`, only when the C preprocessor macro `DEBUG` is defined. It's called from `backend_init` in a `debug` statement in `dmd/dmsc.d`.

**Comment by wilzbach** [comment]
Looks like this one was "healed" by the DMD backend conversion to D (:

File: src/win32.mak [hunk 1]
```diff
@@ -306,7 +306,8 @@ trace:
 	$(DMDMAKE) "OPT=-o" "DEBUG=-gt -Nc" "DDEBUG=-debug -g -unittest" "DOPT=" "LFLAGS=-L/ma/co/delexe/la" $(TARGETEXE)
 
 unittest:
-	$(DMDMAKE) "OPT=-o" "DEBUG=" "DDEBUG=-debug -g -unittest -cov" "DOPT=" "LFLAGS=-L/ma/co/delexe/la" $(TARGETEXE)
+	$(DMDMAKE) "OPT=-o" "DEBUG=" "DDEBUG=-g -unittest -cov" "DOPT=" "LFLAGS=-L/ma/co/delexe/la" $(TARGETEXE)
```
Comment (wilzbach) [inline src/win32.mak]: Debug builds fail also on Linux at the moment: https://github.com/dlang/dmd/pull/7528

**Comment by wilzbach** [comment]
CC @thewilsonator this is ready to go and passing on the CIs.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Auto-merge toggled on

**Comment by thewilsonator** [comment]
Auto-merge toggled off

**Comment by thewilsonator** [comment]
Its failing agin now.

**Review by RazvanN7 — APPROVED** [review]

**Comment by rainers** [comment]
removing auto-merge because the unittest is failing. Running dmd without args is considered an error, something like `dmd --version` should be used instead.

**Comment by MoonlightSentinel** [comment]
This can be closed beceause #10612 enabled unittesting on windows

**Comment by wilzbach** [comment]
Thanks!

**Pull request closed without merging**